### DISCUSSION TITLE
How inject additional correlation field into logs using javaagent extension

### DISCUSSION THREAD

**Discussion by EugeneMsv** [open-telemetry/opentelemetry-java-instrumentation#7832] [Q&A]
Hi, 

There is a list of auto injected correlation fields to app logs
https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/logs#auto-instrumenting-existing-logging 
like `trace_id`. 
Can you pls suggest a place where it is implemented and might be an example how we can add our custom field from otel Context using the same approach? I'm particulary interested in adding the field via otel javaagaent extension, not the application code. 
Thank you.

**Comment by trask** [comment]
hi @EugeneMsv, logback mdc instrumentation now supports injecting baggage into mdc: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/logback/logback-mdc-1.0/javaagent/README.md

**Reply by EugeneMsv** [reply]
Thank you.
I've already used an alternative way with SPI. 
But I'll take a look at your suggestion
